### PR TITLE
Add Exoscale provider

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
This PR add the Exoscale provider

### CODE REVIEW

**Comment by pierre-emmanuelJ** [comment]
⚠️ Execute the test with an organization with 0 compute instances created. Because the first test is executed as a full node organization discovery.

Deployed resource with Terraform to execute test:
```
export EXOSCALE_API_KEY=... && export EXOSCALE_API_SECRET=... && export EXOSCALE_API_ENDPOINT="https://api.exoscale.com/v1"
cd test/tf/exoscale
terraform init
terraform apply
```

then run test from the root repo
```
go test -v ./provider/exoscale
=== RUN   TestAddrs
2020/06/30 11:18:19 [INFO] discover-exoscale: Address type  is not supported. Valid values are public_v4 or public_v6. Falling back to 'public_v4'
2020/06/30 11:18:19 [DEBUG] discover-exoscale: Using zone= tag_key=, tag_value=, addr_type=public_v4
2020/06/30 11:18:19 [DEBUG] discover-exoscale: Found instance test-01 - with public_v4 IP: 89.145.162.37
2020/06/30 11:18:19 [DEBUG] discover-exoscale: Found instance test-02 - with public_v4 IP: 185.19.28.136
2020/06/30 11:18:19 [DEBUG] discover-exoscale: Found ip addresses: [89.145.162.37 185.19.28.136]
--- PASS: TestAddrs (0.27s)
=== RUN   TestAddrsZone
2020/06/30 11:18:19 [INFO] discover-exoscale: Address type  is not supported. Valid values are public_v4 or public_v6. Falling back to 'public_v4'
2020/06/30 11:18:19 [DEBUG] discover-exoscale: Using zone=ch-gva-2 tag_key=, tag_value=, addr_type=public_v4
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Found instance test-02 - with public_v4 IP: 185.19.28.136
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Found ip addresses: [185.19.28.136]
--- PASS: TestAddrsZone (0.33s)
=== RUN   TestAddrsTags
2020/06/30 11:18:20 [INFO] discover-exoscale: Address type  is not supported. Valid values are public_v4 or public_v6. Falling back to 'public_v4'
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Using zone= tag_key=test-01, tag_value=consul, addr_type=public_v4
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Found instance test-01 - with public_v4 IP: 89.145.162.37
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Found ip addresses: [89.145.162.37]
--- PASS: TestAddrsTags (0.23s)
=== RUN   TestAddrsIPv6
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Using zone= tag_key=, tag_value=, addr_type=public_v6
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Found instance test-02 - with public_v6 IP: 2a04:c43:e00:46b0:4f5:92ff:fe00:c5
2020/06/30 11:18:20 [DEBUG] discover-exoscale: Found ip addresses: [2a04:c43:e00:46b0:4f5:92ff:fe00:c5]
--- PASS: TestAddrsIPv6 (0.18s)
PASS
ok  	github.com/hashicorp/go-discover/provider/exoscale	1.054s
```

File: provider/exoscale/exoscale_discover.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
+// Package exoscale provides node discovery for Scaleway.
```
Comment (gabrielexoscale) [inline provider/exoscale/exoscale_discover.go]: Wanna remove the Scaleway reference

File: provider/exoscale/exoscale_discover.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    tag_key:        The tag key to filter on
+    tag_value:      The tag value to filter on
+    zone:           The Exoscale Zone
+    addr_type:      "public_v4" or "public_v6". (default: "public_v4")
+`
+}
+
+func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error) {
+	if args["provider"] != "exoscale" {
+		return nil, fmt.Errorf("discover-scaleway: invalid provider " + args["provider"])
```
Comment (illi-j) [inline provider/exoscale/exoscale_discover.go]:
```suggestion
		return nil, fmt.Errorf("discover-exoscale: invalid provider " + args["provider"])
```

**Review by illi-j — APPROVED** [review]
🏖️

**Review by gabrielexoscale — APPROVED** [review]

**Pull request merged**